client/src/components/auth/Register.js

```diff
@@ -16,6 +18,12 @@ const initialState = {
 class Register extends React.Component {
   state = initialState;
 
+  componentDidMount() {
+    if (this.props.auth.username) {
+      this.setState({ redirectToLogin: true });
```

Comment:
Should be `redirectToDashboard` instead (the user is already logged in).
Effect is still the same though, the user is logged in → try to go to `/register` → they're redirected to `/login` → they're redirected again to `/dashboard`